assessments/projects/knative/self-assessment.md

```diff
@@ -0,0 +1,577 @@
+
+---
+ 
+## Self-assessment Use
+This self-assessment is created by a team of cybersecurity students at NYU to perform an external analysis of the project's security. It is not intended to provide a security audit of Knative, or function as an independent assessment or attestation of Knative's security health.
+
+This document serves to provide Knative users with an initial understanding of Knative's security, where to find existing security documentation, Knative plans for security, and general overview of Knative security practices, both for development of Knative as well as security of Knative.
+
+This document provides the CNCF TAG-Security with an initial understanding of Knative to assist in a joint-assessment, necessary for projects under incubation. Taken together, this document and the joint-assessment serve as a cornerstone for if and when Knative seeks graduation and is preparing for a security audit.
+
+---
+
+## Security functions and features
+| Component | Applicability | Description of Importance |
+| --------- | ------------- | ------------------------- |
+| Code Signature Verification | Security Relevant | Knative signs their releases with cosign to verify binaries, ensuring integrity and authenticity for users. |
+| Istio | Security Relevant | Istio is an open platform-independent service mesh that provides traffic management, policy enforcement, and telemetry collection. Although Istio is not required for Knative to operate, it can be used for networking operations like traffic rules, authorization policies, and encryption. |
+| Kubernetes Infrastructure | Critical | Knative is built upon Kubernetes, which is configured with security tools like RBAC, network policies, and secrets. |
+| Security Guard | Security Relevant | Security Guard is an extension that helps secure microservices, serverless containers and serverless functions. It detects and blocks exploits sent to services and can detect and restart compromised service pods. May not be widely deployed. |
+
+---
+
+## Project Compliance
+Not enough information could be gathered to determine exactly what regulatory standards Knative itself complies with. However, Knative is built on Kubernetes, which follows the following regulatory standards depending on what industry it is deployed in: SOC 2, PCI DSS, HIPAA, NIST SP 800-53, GDPR, ISO 27001. It could be said that Knative follows the CNCF Community Code of Conduct and the CNCF IP Policy. This should be easily accessible information and should be considered in the documentation.
+
+---
+
+## Secure Development Practices
+The Knative project is still incubating. The security policies and development practices detailed below are based on open source guidelines and preliminary threat modeling.
+
+### Deployment Pipeline
+
+Knative’s release process is mostly automated. Following is a chronological description of the deployment, sourced from https://github.com/knative/release/blob/main/PROCEDURES.md
+
+1. Open PR check  
```

Comment:
Are there any additional security checks performed in this pipeline, like container image scan, SAST or secret scanning? Please include if any